callstats-java-sdk/src/main/java/io/callstats/sdk/httpclient/CallStatsHttp2Client.java

```diff
@@ -0,0 +1,138 @@
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class CallStatsHttp2Client {
+	
+	/** The base url. */
+	private String EVENTS_BASE_URL;
+	private String STATS_BASE_URL;
+	private String AUTH_BASE_URL;
+	
+	/** The connection time out. */
+	private int connectionTimeOut = CallStatsConst.CONNECTION_TIMEOUT;
+	
+	/** The Constant logger. */
+	private static final Logger logger = LogManager.getLogger("CallStatsAsyncHttpClient");
+	
+	private OkHttpClient okHttpClientclient;
+	private boolean isDisrupted;
+	
+	public boolean isDisrupted() {
+		return isDisrupted;
+	}
+
+	public void setDisrupted(boolean isDisrupted) {
+		this.isDisrupted = isDisrupted;
+	}
+	
+	public CallStatsHttp2Client() {
+		okHttpClientclient = new OkHttpClient.Builder()
+			.readTimeout(connectionTimeOut, TimeUnit.MILLISECONDS)
+			.writeTimeout(connectionTimeOut, TimeUnit.MILLISECONDS)
+			.connectionPool(new ConnectionPool())
+			.build();
+		loadConfigurations();
+	}
+	
+	private void loadConfigurations() {
+		Properties prop = new Properties();
+		InputStream input = null;
+		
+		try {
+			input = new FileInputStream(CallStatsConst.CallStatsJavaSDKPropertyFileName);
+			if (input !=null){
+				prop.load(input);
+				EVENTS_BASE_URL = prop.getProperty("CallStats.EventsBaseURL");
+				STATS_BASE_URL = prop.getProperty("CallStats.StatsBaseURL");
+				AUTH_BASE_URL = prop.getProperty("CallStats.AuthBaseURL");
+				
+				if (prop.getProperty("CallStats.ConnectionTimeOut") != null) {
+					connectionTimeOut = Integer.parseInt(prop.getProperty("CallStats.ConnectionTimeOut"));
+				}
+				
+				if (EVENTS_BASE_URL == null || STATS_BASE_URL == null) {
+					logger.error("Callstats BASE URL can not be null");
+					throw new RuntimeException("Callstats Events and Stats BASE URL can not be null");
+				}
+			}
+			logger.info("Base URL is " + EVENTS_BASE_URL);
+		}  catch (FileNotFoundException e ) {
+			logger.error("Configuration file not found", e);
+			throw new RuntimeException("Configuration file not found");
+		} catch (IOException e ) {
+			logger.error("Configuration file read IO exception", e);
+			throw new RuntimeException("Configuration file read IO exception");
+		}
+	}
+	
+	public void sendHttp2Request(String url, String body, final CallStatsHttp2ResponseListener listener) {
+		MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+	    RequestBody msg = RequestBody.create(JSON, body);
+	    Request request = new Request.Builder()
+        .url(EVENTS_BASE_URL+url)
+        .addHeader("Accept", "application/json")
+	    .addHeader("Accept-Charset", "utf-8")
+        .post(msg)
+        .build();
+	    logger.debug("sending request "+body);
+	    send(request, listener);
+	}
+	
+	public void sendHttp2Request(String url, List<NameValuePair> paramList, final CallStatsHttp2ResponseListener listener) {
```

Comment:
Can you rename these two something more descriptive? They are more something like sendEventRequest and sendAuthRequest